shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/WADODownloaderService.java

```diff
@@ -425,9 +422,16 @@ private void extractDICOMFilesFromMHTMLFile(final byte[] responseBody, final Str
 	 * @throws IOException
 	 * @throws MessagingException
 	 */
-	private void extractDICOMZipFromMHTMLFile(final byte[] responseBody, String name, ZipOutputStream zipOutputStream)
+	private void extractDICOMZipFromMHTMLFile(final byte[] responseBody, String name, ZipOutputStream zipOutputStream, boolean isMultipart)
 			throws IOException, MessagingException {
 		try(ByteArrayInputStream bIS = new ByteArrayInputStream(responseBody)) {
+			if (!isMultipart) {
```

Comment:
I think this method should be now structured like this :

```
// not multipart
if (!isMultipart) {
   ...
   return;
}

...
// multipart with 1 file
if (count == 1) {
   // invert condition here
   if (!(bodyPart.isMimeType(CONTENT_TYPE_DICOM) && bodyPart.isMimeType(CONTENT_TYPE_DICOM_XML))) {
      throw new IOException(...);
   }
   ...
   return
}

// multipart with multiple file
for (int i = 0; i < count; i++) {
   // invert condition here
   if (!(bodyPart.isMimeType(CONTENT_TYPE_DICOM) && bodyPart.isMimeType(CONTENT_TYPE_DICOM_XML))) {
      throw new IOException(...);
   }
   ...
}
```

Would clarify cases, and eliminate a bit of cyclomatic complexity